src/main/scala/com/datastax/sparkstress/WriteTask.scala

```diff
@@ -47,20 +56,54 @@ abstract class WriteTask[rowType](
 
   def getRDD: RDD[rowType]
 
-  def run() = {
-    config.saveMethod match {
-      case "bulk" => getRDD.bulkSaveToCassandra(config.keyspace, config.table)
-      case _ => new RDDFunctions(getRDD).saveToCassandra(config.keyspace, config.table)
-      //For Some reason the implicit doesn't work here in Connector 1.[1,0].X
-  	}
+  /**
+   * Runs the write workload, returns when terminationTimeMinutes is reached or when the job completes, which ever is first.
+   * @return a tuple containing (runtime, totalCompletedOps)
+   */
+  def run(): (Long, Long) = {
+    var totalCompletedOps: Long = 0L
+    val runtime = time({
+      val fs = Future {
+        config.saveMethod match {
+          case "bulk" => getRDD.bulkSaveToCassandra(config.keyspace, config.table)
+          case _ => new RDDFunctions(getRDD).saveToCassandra(config.keyspace, config.table)
+          //For Some reason the implicit doesn't work here in Connector 1.[1,0].X
+        }
+      }
+      try {
+        if (config.terminationTimeMinutes > 0) {
+          Await.result(fs, Duration(TimeUnit.MINUTES.toMillis(config.terminationTimeMinutes), MILLISECONDS))
+        } else {
+          Await.result(fs, Duration(Long.MaxValue, NANOSECONDS)) // max allowed duration 292 years :)
+        }
+      } catch {
+        case ex: TimeoutException => {
+          println(s"We hit our timeout limit for this test (${config.terminationTimeMinutes} min), shutting down.")
+          val host: String = sys.props.get("spark.master").getOrElse(sc.getConf.getOption("spark.master")).toString.split("://|:")(1)
+          val appId: String = sc.getConf.getAppId
+
+          /* Follow-up: For now we're only pulling stage id 0, if/when we want to support multiple trials we may want to
+          pass in the trial number to run() and use that as the stage id. */
+          val stage0: Option[JSONType] = JSON.parseRaw(IOUtils.toString(new URL(s"http://${host}:4040/api/v1/applications/${appId}/stages/0")))
```

Comment:
Hmm I don't see a path for extracting outputRecords. I see sc.[statusTracker](https://spark.apache.org/docs/1.6.2/api/java/org/apache/spark/SparkContext.html#statusTracker()).[getStageInfo](https://spark.apache.org/docs/1.6.2/api/java/org/apache/spark/SparkStatusTracker.html#getStageInfo(int))(0) from where we can get task info. I'll keep sniffing around.